src/test/java/ssl/ois/timelog/ut/service/team/GetRoleUseCaseTest.java

```diff
@@ -0,0 +1,126 @@
+package ssl.ois.timelog.ut.service.team;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import ssl.ois.timelog.service.exception.role.GetRoleErrorException;
+import ssl.ois.timelog.service.repository.user.UnitRepository;
+import ssl.ois.timelog.service.role.get.*;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ssl.ois.timelog.adapter.repository.memory.MemoryUnitRepository;
+import ssl.ois.timelog.model.connect.UnitInterface;
+import ssl.ois.timelog.model.team.Role;
+import ssl.ois.timelog.model.team.Team;
+import ssl.ois.timelog.model.user.User;
+
+public class GetRoleUseCaseTest {
+    private UnitRepository unitRepository;
+    private GetRoleUseCaseInput input;
+    private GetRoleUseCaseOutput output;
+    private GetRoleUseCase getRoleUseCase;
+    private Map<String, UnitInterface> users;
+    private Map<UUID, Role> memberRoleMap;
+    private List<UnitInterface> teams;
+    private UUID memberID;              //8e0f3043-85d7-4c7b-9ac9-cfb1149270b9
+    private UUID leaderID;              //5a39fabf-f969-4dfb-8878-f1b1bbe7188b
+    private UUID professorID;           //7be599c9-0cb1-42f6-9ec3-95bfc446c42e
+    private UUID stakeHolderID;         //d5fedb30-1489-41da-ba91-0211431d5ef3 
+    private UUID teamID;                //5edd79d6-7d15-4e97-b6b4-fb8ed76e47ab
+
+    @Before
+    public void setup(){
+        input = new GetRoleUseCaseInput();
+        output = new GetRoleUseCaseOutput();
+        this.unitRepository = new MemoryUnitRepository();
+        getRoleUseCase = new GetRoleUseCase(unitRepository);
+        this.users = new HashMap<>();
+        this.memberRoleMap = new HashMap<>();
+        this.teams = new ArrayList<>();
+
+        memberID = UUID.fromString("8e0f3043-85d7-4c7b-9ac9-cfb1149270b9");
+        leaderID = UUID.fromString("5a39fabf-f969-4dfb-8878-f1b1bbe7188b");
+        professorID = UUID.fromString("7be599c9-0cb1-42f6-9ec3-95bfc446c42e");
+        stakeHolderID = UUID.fromString("d5fedb30-1489-41da-ba91-0211431d5ef3");
+        teamID = UUID.fromString("5edd79d6-7d15-4e97-b6b4-fb8ed76e47ab");
+
+        users.put(memberID.toString(), new User(memberID));
+        users.put(leaderID.toString(), new User(leaderID));
+        users.put(professorID.toString(), new User(professorID));
+        users.put(stakeHolderID.toString(), new User(stakeHolderID));
+
+        memberRoleMap.put(memberID, Role.MEMBER);
+        memberRoleMap.put(leaderID, Role.LEADER);
+        memberRoleMap.put(professorID, Role.PROFESSOR);
+        memberRoleMap.put(stakeHolderID, Role.STAKEHOLDER);
+
+        teams.add(new Team(teamID, memberRoleMap));
+
+        ((MemoryUnitRepository)this.unitRepository).init(this.users, this.teams);
```

Comment:
this.unitRepository = new MemoryUnitRepository(this.users,this.teams);

then add new constructor at MemoryUnitRepository